### ISSUE TITLE
VPN before Windows logon

### ISSUE THREAD

**Issue by LokoOn** [netbirdio/netbird#2809]
**Is your feature request related to a problem? Please describe.**
It's not uncommon for a Windows domain-joined device to be located outside of its home network.

Often, such a device will have an internet connection before a user logs in. If a VPN connection could be established prior to user login, settings, updates, and other configurations from the domain controller could be synchronized directly from the LAN, even without an active user session on the Windows device.

Additionally, with an established VPN connection, login credentials could be verified directly against the central directory. This would also enable users to log in to a Windows client for the first time, even if they had never previously logged in on that particular device.

For these reasons, a pre-login VPN connection presents a valuable solution.

**Describe the solution you'd like**
A pre-login VPN connection at the User logon screen.

**Comment by Lamera** [comment]
@LokoOn Is this not actually the case when you deploy netbird with a setup key?

**Comment by LokoOn** [comment]
@Lamera you are totally right. When using a setup, than the VPN is established automatically on system start.
That's handy for container, server workloads.
I need to add some to the feature request.

There should be an option to establish a NetBird connection before the Windows login using user authentication. This ensures that only an authorized user can create a VPN connection. Using a setup key is not a viable solution, as it authenticates the machine rather than the user, allowing any user on the device to establish a NetBird connection. This could pose a security risk in the case of device loss

**Comment by snailzrus** [comment]
100% agree, this would be a really great addition for using Netbird for corporate devices so that they can auth the windows login against an Active Directory server elsewhere on Earth

**Comment by snailzrus** [comment]
So Windows makes it possible to establish a VPN connection before Windows login has been completed. It presents authentication to the user, so there shouldn't be any additional risk. I think the biggest problem I can think of is how Windows allows that auth, which is that Windows likely won't allow a browser SSO auth before the user has logged in.

Here is the doc that Palo Alto has for this feature, though it's not using SSO auth:
https://docs.paloaltonetworks.com/globalprotect/5-2/globalprotect-app-user-guide/globalprotect-app-for-windows/use-connect-before-logon-followed-by-the-authentication-method

**Comment by Gauss23** [comment]
I think using a setup key in combination with Bitlocker and a pre-boot password would be also an option. In case of device loss the the new owner is not able to do very much.

**Comment by Matt0550** [comment]
Any news/workaround?

**Comment by iFrancisMe** [comment]
I am interested in this functionality as well. Your product is intriguing as I have been looking for something like this, but I have yet to try it out. Anyways, I think a solution to this would be as mlsmaycon suggested. Perhaps issuing a key with limited access to a targeted list of allowed endpoints (domain controllers), then allow Windows to authenticate the user via Active Directory. I assume that most Active Directory environments interested in this would have also implemented Microsoft's Entra ID sync along with seamless sso so that authenticating in Windows automatically authenticates via Azure. Then once authenticated via Azure, the Netbird client can authorize full vpn access. Couldn't that work?

**Comment by BiasF** [comment]
On PaloAlto, the PreLogonTunnel is established based on a computer certificate or a "cookie" in the GlobalProtect Client. 
The benefit of the PreLogon tunnel, is that software deployment systems can work before the user is logged in. 
And also for classic group policies in a active directory environment a connection to a domain controller is needed. For some GPOs it's essential that the GPO is applied BEFORE user login. 

After logging in, the PreLogonTunnel is terminated and the user tunnel with specific user rights is connected.
